components/infra-proxy-service/migrations/pipeline/utility.go

```diff
@@ -305,39 +308,169 @@ func Unzip(ctx context.Context, mst storage.MigrationStorage, result pipeline.Re
+	log.Info("starting with user parseing phase for migration id: ", result.Meta.MigrationID)
+
+	file := path.Join(result.Meta.UnzipFolder, "key_dump.json")
+
+	keyDumpByte, err := ioutil.ReadFile(file)
+	if err != nil {
+		log.Errorf("failed to read keydump file for user parsing for the migration id: %s : %s", result.Meta.MigrationID, err.Error())
+		return result, err
+	}
+
+	var keyDumps []pipeline.KeyDump
+	if err := json.Unmarshal(keyDumpByte, &keyDumps); err != nil {
+		log.Errorf("failed to unmarshal for user parsing for the migration id: %s : %s", result.Meta.MigrationID, err.Error())
+		return result, err
+	}
+
+	serverUsers := keyDumpTOUser(keyDumps)
+	automateUsers, err := st.GetAutomateInfraServerUsers(ctx, result.Meta.ServerID)
+	if err != nil {
+		log.Errorf("failed to get users form db for user parsing for the migration id: %s : %s", result.Meta.MigrationID, err.Error())
+		return result, err
+	}
+
+	var mappedUsers []pipeline.User
+
+	mappedUsers = append(mappedUsers, deleteUser(serverUsers, automateUsers)...)
+	mappedUsers = append(mappedUsers, insertUpdateSkipUser(serverUsers, automateUsers)...)
+
+	result.ParsedResult.Users = mappedUsers
+	return result, nil
+}
+
+// Clean serialized_object and Polulate Users struct
+func keyDumpTOUser(keyDump []pipeline.KeyDump) []pipeline.User {
+	var users []pipeline.User
+	for _, kd := range keyDump {
+		sec := map[string]string{}
+		if err := json.Unmarshal([]byte(kd.SerializedObject), &sec); err != nil {
+			log.Errorf("failed to pasre user's first, middle and last name: %s", err.Error())
+		}
+		user := pipeline.User{
+			Username:    kd.Username,
+			Email:       kd.Email,
+			DisplayName: sec["display_name"],
+			FirstName:   sec["first_name"],
+			LastName:    sec["last_name"],
+			MiddleName:  sec["middle_name"],
+		}
+		users = append(users, user)
+	}
+	return users
+}
+
+func automateMap(automateUser []storage.User) map[string]storage.User {
+	autoMap := map[string]storage.User{}
+	for _, auser := range automateUser {
+		autoMap[auser.InfraServerUsername] = auser
+	}
+	return autoMap
+}
+
+func serverMap(server []pipeline.User) map[string]pipeline.User {
+	autoMap := map[string]pipeline.User{}
+	for _, auser := range server {
+		autoMap[auser.Username] = auser
+	}
+	return autoMap
+}
+
+func insertUpdateSkipUser(serverUser []pipeline.User, automateUser []storage.User) []pipeline.User {
+	var parsedUsers []pipeline.User
+	autoMap := automateMap(automateUser)
+	for _, sUser := range serverUser {
+
+		if autoMap[sUser.Username].InfraServerUsername != "" {
+			emptyVal := pipeline.User{}
+			returnedVal := skipOrUpdate(autoMap, sUser)
+			if returnedVal != emptyVal {
+				parsedUsers = append(parsedUsers, returnedVal)
+			}
+		} else {
+			if sUser.Username == "pivotal" {
+				sUser.ActionOps = pipeline.Skip
+				parsedUsers = append(parsedUsers, sUser)
```

Comment:
Instead of adding pivotal user to the skip, we can directly continue on the loop